### PR TITLE
Update node and fix search lag

### PR DESCRIPTION
**Pull request by afaur**
Currently when typing into the input of zazu there is a delay caused because a fn passed down through the many parent components.  

When this fn is called (each time that the search component input detects a change) we currently determine which plugins have matched results and then additionally pass the new search input value back down to the search component.

This change will cut off the usage of the passed down search input value. It will instead set a local state value of searchQuery that at present isn't even bound to the value prop on the input (as I am unsure if it is actually needed?).

The other part of this work is the inclusion of a debounce on the handleQueryChange fn that will throttle the detection of which plugins matched when you type or remove characters in the input box.

I also included a node version bump to see if travis and appveyor is able to handle the latest node release.

### CODE REVIEW

File: .nvmrc [hunk 1]
```diff
@@ -1 +1 @@
-v9.8.0
+v10.0.0
```
Comment (bayleedev) [inline .nvmrc]: `v9.8.0` is the node version that the current electron version uses. They should stay in sync.

File: app/components/search.js [hunk 2]
```diff
@@ -94,65 +85,48 @@ class Search extends React.Component {
...
-  componentWillUnmount = () => {
-    globalEmitter.removeListener('hideWindow', this.handleSaveQuery)
-    keyboard.unbind('search')
-  }
 
-  componentDidUpdate = () => {
-    if (this.props.value === '') {
-      this.focus()
+    // https://stackoverflow.com/questions/23123138/perform-debounce-in-react-js#answer-24679479
+    this.handleQueryChange = debounce(_ => this.props.handleQueryChange(this.state.searchQuery), 100)
```
Comment (bayleedev) [inline app/components/search.js]: Debounce is handled by the plugin blocks. Both root and prefix have a debounce.

http://zazuapp.org/documentation/blocks/#root-script

Maybe giving this a default for those who don't set it?

File: app/lib/debounce.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
+// https://github.com/component/debounce/blob/master/index.js
+module.exports = (func, wait, immediate) => {
```
Comment (bayleedev) [inline app/lib/debounce.js]: This debouncer uses the task queue, not the microtask queue like the one in the blocks is implemented.

**Review by bayleedev — CHANGES_REQUESTED** [review]

File: app/components/search.js (hunk 2 above)
Comment (afaur) [reply app/components/search.js]: I gather you would like this to handle debounce 
(this is just an extract and refactor of the debounce you mentioned):
https://gist.github.com/afaur/ff126a0c2d016e3b7fa66f679012c599

When you say "giving this a default" do you mean: 
Set up a zazu config option for the amount of time to deflect successive fn calls?

File: .nvmrc (hunk 1 above)
Comment (afaur) [reply .nvmrc]: Makes sense.

**Comment by linonetwo** [comment]
The problem seems to be resolved in 0.6.0

**Pull request closed without merging**